packages/xsnap/src/xsnap.js

```diff
@@ -312,7 +312,12 @@ export async function xsnap(options) {
         xsnapProcess.kill();
         throw Error('xsnap protocol error: received empty message');
       } else if (message[0] === OK) {
-        let meterInfo = { compute: null, allocate: null, timestamps: [] };
+        let meterInfo = {
+          compute: null,
+          allocate: null,
+          currentHeapCount: null,
+          timestamps: [],
+        };
```

Comment:
Help me understand the upgrade considerations? How should I expect this change to get deployed?

IIUC, the norm these days is to build an a3p test that verifies the impact of changes like this in a running blockchain. We might decide that's not cost-effective here, but I'd like to think through what it would look like.

Reply:
This file is executed as part of the kernel, so it gets deployed when we do a chain-halting upgrade that changes the kernel code we're running.

This shouldn't have any negative effects, in fact I don't think the kernel behavior will change at all, see the other comment.

Reply:
ah. this is kernel side, not in any vats.